libraries/chain/controller.cpp

```diff
@@ -1246,14 +1246,49 @@ struct controller_impl {
       }
    }
 
+   template<typename ForkDB, typename BSP>
+   void apply_irreversible_block(ForkDB& forkdb, const BSP& bsp) {
+      if( read_mode == db_read_mode::IRREVERSIBLE ) {
```

Comment:
You can do a `if( read_mode != db_read_mode::IRREVERSIBLE ) return;` to avoid one level of nesting. But this is minor.